X10D/src/Text/RuneExtensions.cs

```diff
@@ -44,17 +46,42 @@ public static string Repeat(this Rune value, int count)
                 return value.ToString();
         }
 
-        int utf8SequenceLength = value.Utf8SequenceLength;
-        Span<byte> utf8 = stackalloc byte[utf8SequenceLength];
-        value.EncodeToUtf8(utf8);
-
-        Span<byte> buffer = stackalloc byte[utf8.Length * count];
-        for (var index = 0; index < count; index++)
+        // Helpful documentation: https://en.wikipedia.org/wiki/UTF-8
+        switch (value.Utf8SequenceLength)
         {
-            utf8.CopyTo(buffer.Slice(index * utf8.Length, utf8.Length));
-        }
+            case 1:
+                {
+                    Unsafe.SkipInit(out byte bytes);
+                    value.EncodeToUtf8(MemoryMarshal.CreateSpan(ref bytes, 1));
+
+                    return new string((char)value.Value, count);
+                }
+
+            case 2:
```

Comment:
Why does case 2 exist? 🤔

Reply:
I'll reimplement this one, it was implemented after a morning wake and I just forgot about it. Looking back at it was a total disaster.